src/libasr/codegen/wasm_to_x86.cpp

```diff
@@ -154,6 +154,41 @@ class X86Visitor : public WASMDecoder<X86Visitor>,
         m_a.asm_push_r32(X86Reg::eax);
     }
 
+    void handle_I32Compare(const std::string &compare_op) {
+        unique_id.push_back(std::to_string(offset));
+        m_a.asm_pop_r32(X86Reg::ebx);
+        m_a.asm_pop_r32(X86Reg::eax);
+        m_a.asm_cmp_r32_r32(X86Reg::eax, X86Reg::ebx);
+        if (compare_op == "Eq") {
+            m_a.asm_je_label(".compare_1" + unique_id.back());
+        } else if (compare_op == "Gt") {
+            m_a.asm_jg_label(".compare_1" + unique_id.back());
+        } else if (compare_op == "GtE") {
+            m_a.asm_jge_label(".compare_1" + unique_id.back());
+        } else if (compare_op == "Lt") {
+            m_a.asm_jl_label(".compare_1" + unique_id.back());
+        } else if (compare_op == "LtE") {
+            m_a.asm_jle_label(".compare_1" + unique_id.back());
+        } else if (compare_op == "NotEq") {
+            m_a.asm_jne_label(".compare_1" + unique_id.back());
+        } else {
+            throw CodeGenError("Comparison operator not implemented");
+        }
+        m_a.asm_mov_r32_imm32(X86Reg::eax, 0);
+        m_a.asm_jmp_label(".compare.end_" + unique_id.back());
+        m_a.add_label(".compare_1" + unique_id.back());
+        m_a.asm_mov_r32_imm32(X86Reg::eax, 1);
+        m_a.add_label(".compare.end_" + unique_id.back());
+        m_a.asm_push_r32(X86Reg::eax);
```

Comment:
In the last discussion, you said we can push value directly into the stack.
Here why not we just push the result value into the stack instead of moving it into the eax?
Also, I have another doubt:
How do we access the pushed value in the stack?

Reply:
> https://github.com/lcompilers/lpython/issues/1222#issuecomment-1305148759
How to perform that?

Reply:
Pushing the value directly into the stack works:
```c++
m_a.asm_push_imm8(0);
m_a.asm_jmp_label(".compare.end_" + std::to_string(offset));
m_a.add_label(".compare_1" + std::to_string(offset));
m_a.asm_push_imm8(1);
m_a.add_label(".compare.end_" + std::to_string(offset));
```

Reply:
>How do we access the pushed value in the stack?

It can be accessed during run-time only.

>How would we use the value on top of stack?

Other instructions can use it during run-time time. They perform some operation(s) on it and store the result back onto stack.

Reply:
>Pushing the value directly into the stack works:

If it works, we can use this way. (I think it could be better as it could use (slightly) lesser instructions)

Reply:
I added this change/improvement into this commit
[Refactor: WASM_X86: Directly push value onto stack](https://github.com/lcompilers/lpython/pull/1285/commits/15779646094f2c40110bfe06cd1dfa0041985e30).